### ISSUE TITLE
Unable to run script with parameters

### ISSUE THREAD

**Issue by ghost** [otris/node-documents-scripting#20]
For debugging purposes it may be benefitial to run a script with a parameter.
Right now the `runScript` function only passes the script name. I tried to find an possible input for `sdsConnection.PDClass.callOperation` to pass a parameter, but could not find any (a look at the code of *node-sds* shows, that there are additional parmeters, but it's not clear on first glance, how they should be formatted).
While you can use the `sdsConnection.CustomOperations.runScriptOnServer` function for a workaround, it's not so "clean":
```javascript
const serverOperations = require('../out/src/serverOperations');
const config = require('../out/src/config');

(async () => {
  const login = new config.ConnectionInformation();
  // login.[…] = […]
  const systemUser = '…';
  const scriptName = '…';
  const scriptParameters = { a: 5 };

  await serverOperations.serverSession(login, [], async (sdsConnection) => {
    let code = `var call = new ScriptCall('${systemUser}', '${scriptName}', true);\n`;

    code += Object.entries(scriptParameters)
      .map(([name, value]) => `call.addParameter('${name}', '${value}');`)
      .join('\n');

    code += `if (call.launch()) { call.waitForFinish(); return call.getReturnValue(); } else { return call.getLastError(); }`;

    console.log(await sdsConnection.CustomOperations.runScriptOnServer(code));
  });
})();
```
One limitation is, that the returnValue seems to be cast to a string inside `runScriptOnServer`. This means that objects become `[object Object]` instead of a JSON. 
Another limitation is, that you have to specify a systemUser which can not be the admin-user with which you login to the session.
And, of course, you only get the return value, while `runScript`/`sdsConnection.PDClass.callOperation` returns more information.

**Comment by tineheller** [comment]
If you want to run a script with parameters, you can only use the function `runScript`. And you need to set the parameters in a special JSON file and upload this file with `uploadScript` before. If you use the VS Code plugin, you can set `vscode-janus-debug.scriptParameters = true` and download a script, then you will get an example JSON file that contains script parameters.
I hope this will help, and I'm sorry there isn't a better way to use parameters.

**Comment by ghost** [comment]
Correct me if I'm wrong, but the way you describe only **configures** the parameter in Documents **without passing it**.
While the JSON contains a parameter-key "value", this seems to represent the default value this parameter should have and not the value I want to pass once.
Example:
I have a script with one parameter "a" and a default value of "42" – all configured in Documents without this package.
How could I debug this script with the parameter "a" having the value "5" without changing the default parameter value?

**Comment by tineheller** [comment]
I'm sorry, you're right. And maybe it's possible to use `runScriptOnServer`.
Have you tried `runScriptOnServer(code, ["paramName", "paramValue"])`?

**Comment by ghost** [comment]
While that does work, it does not help with my problem. 
Let me rephrase my goal: I want to execute a script, which is already uploaded to Documents. I don't want/need to run arbitrary code. The whole `runScriptOnServer`-code with the `ScriptCall` in my first post is only a workaround to achieve this. 

Therefore, what I really want is to use this function with parameters: https://github.com/otris/node-documents-scripting/blob/0e41770b449eea6317f33e2582cc42307b129920/src/serverOperations.ts#L1197
What I tried and didn't work:
```javascript
// throws error in term_utf8 from node-sds:
sdsConnection.PDClass.callOperation('PortalScript.runScript', [script.name], ["paramName", "paramValue"]) 
// does nothing (ReferenceError for paramName in Documents still there)
sdsConnection.PDClass.callOperation('PortalScript.runScript', [script.name, "paramName", "paramValue"]) 
// does nothing (ReferenceError for paramName in Documents still there)
sdsConnection.PDClass.callOperation('PortalScript.runScript', [script.name], {"paramName": "paramValue"}) 
// does nothing (ReferenceError for paramName in Documents still there)
sdsConnection.PDClass.callOperation('PortalScript.runScript', [script.name], {0: "paramName", 1: "paramValue"}) 
```
The only thing that worked was using my workaround (without `callOperation`) with the mentioned caveats.

**Comment by tineheller** [comment]
Ok, you can only give one single array with arguments to `callOperation`. So this is the only way, it could work:
```
sdsConnection.PDClass.callOperation('PortalScript.runScript', [script.name, "paramName", "paramValue"]) 
```
If it doesn't, I'll have to take a closer look later...

**Comment by ghost** [comment]
Hi, sadly this did not work when I tried it.
Thanks for your effort!

**Comment by tineheller** [comment]
I debugged into the code this morning. Unfortunately, it's not possible to set script parameters via callOperation(runScript...).
